dashboard/src/main/kotlin/com/ritense/dashboard/datasource/WidgetDataSourceResolver.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2015-2023 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.dashboard.datasource
+
+import org.springframework.boot.context.event.ApplicationStartedEvent
+import org.springframework.context.event.EventListener
+import java.lang.reflect.Method
+
+class WidgetDataSourceResolver : AnnotatedClassResolver() {
+
+    @EventListener(ApplicationStartedEvent::class)
+    fun loadWidgetDataSources() {
+        findWidgetDataSourceClasses()
```

Comment:
We're not adding classes to the classpath at runtime, so instead of using an EventListener, we could just use `init { findWidgetDataSourceClasses() }`

Reply:
changed